### DISCUSSION TITLE
How to generate response spectrum with standards ?

### DISCUSSION THREAD

**Discussion by IcareIbnFirnas** [Dlubal-Software/RFEM_Python_Client#309] [Q&A] [answered]
Hello, I am automatising some computations and I need to generate automatically with Python the response spectrum with the standards of EN 1998-1 of the Eurocodes (as you can see on the pictures).
How can I do it ?
![image](https://github.com/Dlubal-Software/RFEM_Python_Client/assets/148334709/b5f6551e-b2a2-45ac-96d9-4d4875014f1a)
And how can I choose the values of the type of the ground, ... with Python ?
![image](https://github.com/Dlubal-Software/RFEM_Python_Client/assets/148334709/d97222c7-057e-446f-965d-a6b78140343e)

Thank you very much for your help :)

**Comment by heetrojivadiya** [comment, marked as answer]
Hello @IcareIbnFirnas,

Thank you for reaching out to us. 

For 1st question: you can set the response spectrum with the standard by an additional parameter to the ResponseSpectrum() class. For example:

```py
from RFEM.DynamicLoads.responseSpectrum import ResponseSpectrum
ResponseSpectrum(1, params={'definition_type':  'ACCORDING_TO_STANDARD'})
```

For 2nd question: Unfortunately, right now it is not possible to set the ground type, importance class, etc. by Python. We will try to add this feature as soon as possible. We will let you know when it will be implemented. 

Thanks and best regards,
Heet Rojivadiya
